File: epers-front-end/src/app/models/nomenclatoare/NCompartimente.ts
export interface NCompartimente {
    id: number;
    denumire: string;
    id_Locatie: number | null;
    data_in: string | null;
    data_sf: string | null;
    sus: number | null;
    jos: number | null;
    subCompartiment: string;
    activ: boolean;
    firma: string;
    idFirma: number | null;
}

export interface NCompartimentDisplay extends NCompartimente {
    locatie: string | null;
}

export interface CompartimentSelection {
    id: number;
    denumire: string;
    idFirma: number | null;
}

File: EpersBackend/Epers.Models/Nomenclatoare/NCompartimente.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Epers.Models.Nomenclatoare
{
    public class NCompartimente
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public string Denumire { get; set; } = string.Empty;

        public int? IdFirma { get; set; }
        public int? Id_Locatie { get; set; }
        public DateTime? Data_in { get; set; }
        public DateTime? Data_sf { get; set; }
        public int? Sus { get; set; }
        public int? Jos { get; set; }
        public string? SubCompartiment { get; set; }

        [NotMapped]
        public bool Activ { get; set; }

        [NotMapped]
        public string Firma { get; set; } = string.Empty;

    }
}

